spotbugsTestCases/src/java/ghIssues/Issue2258.java

```diff
@@ -0,0 +1,24 @@
+package ghIssues;
+
+public class Issue2258 {
+	private boolean isHole;
+	private Issue2258 next;
+	
+	public void fixHoleLinkage() {
+		Issue2258 orfl = next;
+		
+		while (orfl != null && (orfl.isHole == isHole)) {
+			orfl = orfl.next;
+		}
+		
+		next = orfl;
```

Comment:
Without the change in that PR, the state of the stack here is the same as if the while loop above wasn't there.